datafusion/substrait/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -1464,3 +1464,13 @@ async fn create_all_type_context() -> Result<SessionContext> {
 
     Ok(ctx)
 }
+
+#[tokio::test]
+async fn try_cast_decimal_to_int() -> Result<()> {
+    roundtrip("SELECT * FROM data WHERE a = TRY_CAST(b AS int)").await
+}
+
+#[tokio::test]
+async fn try_cast_decimal_to_string() -> Result<()> {
+    roundtrip("SELECT * FROM data WHERE a = TRY_CAST(b AS string)").await
+}
```

Comment:
minor: There's some existing tests that verify other parts of casting behaviour (`cast_decimal_to_int()`, `implicit_cast()`. Could you move these closer to those for organizational purposes. The test themselves look great ✨